codejail/jail_code.py

```diff
@@ -210,10 +211,9 @@ def jail_code(command, code=None, files=None, extra_files=None, argv=None,
         if user:
             # Run as the specified user
             cmd.extend(['sudo', '-u', user])
+            cmd.extend('{}="{}"'.format(var, env[var]) for var in env)
             rm_cmd.extend(['sudo', '-u', user])
 
-        # Point TMPDIR at our temp directory.
-        cmd.extend(['TMPDIR=tmp'])
```

Comment:
I'm not sure what would happen if you used the absolute path as TMPDIR.  As I said, a lot of this was trial-and-error.